### PR TITLE
add new demo for kubernetes with libstorage

### PR DESCRIPTION
**Pull request by cantbewong**
created new lab to demo kubernetes with libstorage integration on AWS

### CODE REVIEW

File: demo-kubernetes-with-libstorage/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Exploring Kubernetes with libstorage persistent volumes on AWS - the Hard Way
```
Comment (vladimirvivien) [inline demo-kubernetes-with-libstorage/README.md]: 😃  What's the easy way ?

File: demo-kubernetes-with-libstorage/README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
```
+
`cluster/kubectl.sh create –f pod-sc-pvc.yml`
+
+## Contribution
+
+Create a fork of the project into your own repository. Make all your necessary changes and create a pull request with a description on what was added or removed and details explaining the changes in lines of code. If approved, project owners will merge it.
+
+
+## Support
```
Comment (vladimirvivien) [inline demo-kubernetes-with-libstorage/README.md]: Steve, `glide` is not really required (at least not for the Kubernetes codebase)

**Review by vladimirvivien — COMMENTED** [review]
Step 6 of the instructions is not necessary.

File: demo-kubernetes-with-libstorage/README.md (hunk 2 above)
Comment (vladimirvivien) [inline demo-kubernetes-with-libstorage/README.md]: On Step 9, I would show the output of the command `rexray volume create --size ..`

Comment (vladimirvivien) [inline demo-kubernetes-with-libstorage/README.md]: Change this line https://github.com/codedellemc/labs/pull/5/files/4dd2b23a772b2f5a65c496f9f0882ba24fb7fe3b#diff-f513cee62547fb1d0e46cd1f0f078efcL100 to `http://127.0.0.1:7979` .  Did that work for you ?

Comment (vladimirvivien) [inline demo-kubernetes-with-libstorage/README.md]: This line probably can come out https://github.com/codedellemc/labs/pull/5/files/4dd2b23a772b2f5a65c496f9f0882ba24fb7fe3b#diff-f513cee62547fb1d0e46cd1f0f078efcR106 - to lessen confusion (please test though if removed).

File: demo-kubernetes-with-libstorage/README.md (hunk 1 above)
Comment (kacole2) [reply demo-kubernetes-with-libstorage/README.md]: hah. i i'll tone down the words. that's my job. you all just worry about giving me clear instructions

File: demo-kubernetes-with-libstorage/README.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    - an ssh key pair for remote login
+    -  An Access Key for API access
+        - Access Key ID
+        - Secret Access Key 
+
+### Step 1: deploy a VM using an AMI
+
+1. Choose EC2 from the AWS console dashboard, an Launch an instance.
+2. We will use the **Ubuntu Server 16.04 LTS (HVM), SSD Volume Type**. 64-bit. The ami number varies by region. A t2.large instance type is recommended because the Kubernetes build we will be performing can utilize over 4GB of memory. (An m4.large could also be utilized.)
+3. Configure instance details to set the Root volume to 40GB. Other settings can be left as default.
```
Comment (kacole2) [inline demo-kubernetes-with-libstorage/README.md]: @cantbewong @vladimirvivien is there a specific reason why AWS is necessary?

File: demo-kubernetes-with-libstorage/README.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+```
+mkdir -p $GOPATH/src/k8s.io
cd $GOPATH/src/k8s.io
git clone https://github.com/vladimirvivien/kubernetes.git
cd kubernetes
git remote add upstream https://github.com/kubernetes/kubernetes.git
git checkout -b libstorage-take4 origin/libstorage-take4
```
+
### Step 7: Install Rex-Ray
```
Comment (kacole2) [inline demo-kubernetes-with-libstorage/README.md]: this needs to be `cd ~`

File: demo-kubernetes-with-libstorage/README.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
   ebs:
    driver: ebs
ebs:
 accessKey: <my-access-key-here>
 secretKey: <my-secret-key-here>
 region:    <my-aws-region>
 maxRetries: 7
```
+
### Step 8: Build and run Kubernetes
```
Comment (kacole2) [inline demo-kubernetes-with-libstorage/README.md]: getting an error here:

```
ubuntu@ip-172-31-36-124:~$ echo "export GOPATH=$HOME/work" >> /.profile
-bash: /.profile: Permission denied
ubuntu@ip-172-31-36-124:~$ sudo echo "export GOPATH=$HOME/work" >> /.profile
-bash: /.profile: Permission denied
```

Comment (kacole2) [reply demo-kubernetes-with-libstorage/README.md]: directions need to be added if `sudo su` is required

**Review by kacole2 — COMMENTED** [review]
hit em up

**Comment by kacole2** [comment]
i can't add any more comments on the commit. i don't get it. 
1. https://github.com/codedellemc/labs/pull/5/files#diff-f513cee62547fb1d0e46cd1f0f078efcR73 needs `sudo`
2. go doesn't get installed properly following your directions earlier

File: demo-kubernetes-with-libstorage/README.md (hunk 4 above)
Comment (kacole2) [reply demo-kubernetes-with-libstorage/README.md]: actually, we can get rid of this. it's not necessary

File: demo-kubernetes-with-libstorage/README.md (hunk 5 above)
Comment (kacole2) [reply demo-kubernetes-with-libstorage/README.md]: here's how the directions need to look to install go properly:

```
sudo apt-get update && sudo apt-get upgrade
curl https://storage.googleapis.com/golang/go1.6.3.linux-amd64.tar.gz | tar xvz
sudo mv go /usr/local
echo "export PATH=\$PATH:/usr/local/go/bin" >> ~/.profile
echo "export GOPATH=$HOME/work" >> ~/.profile
mkdir $HOME/work
source ~/.profile
```

**Comment by kacole2** [comment]
@cantbewong I'm still having issues following everything. There is definitely some missing steps to be truly spoon-fed. 

here is where it's crashing on me. Directions from the very beginning...:

```
cd ~
curl https://storage.googleapis.com/golang/go1.6.3.linux-amd64.tar.gz | tar xvz
sudo chown -R root:root ./go/
sudo mv go /usr/local
echo "export GOPATH=$HOME/work" >> ~/.profile
echo "export PATH=$PATH:/usr/local/go/bin:$GOPATH/bin" >> ~/.profile
source ~/.profile
sudo apt-get update
sudo apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D
echo "deb https://apt.dockerproject.org/repo ubuntu-xenial main" | sudo tee /etc/apt/sources.list.d/docker.list
sudo apt-get update
sudo apt-get install -y docker-engine
curl -L https://github.com/coreos/etcd/releases/download/v3.0.13/etcd-v3.0.13-linux-amd64.tar.gz | tar xvz
sudo chown -R root:root ./etcd-v3.0.13-linux-amd64/
sudo mv etcd-v3.0.13-linux-amd64/ /usr/local/
sudo ln -s /usr/local/etcd-v3.0.13-linux-amd64/etcd /usr/local/bin/etcd
sudo ln -s /usr/local/etcd-v3.0.13-linux-amd64/etcdctl /usr/local/bin/etcdctl
sudo add-apt-repository ppa:masterminds/glide
sudo apt-get update && sudo apt-get install -y g++ glide make unzip
go get -u github.com/tools/godep
go get -u github.com/jteeuwen/go-bindata/go-bindata
mkdir -p $GOPATH/src/k8s.io
cd $GOPATH/src/k8s.io
git clone https://github.com/vladimirvivien/kubernetes.git
cd kubernetes
git remote add upstream https://github.com/kubernetes/kubernetes.git
git checkout -b libstorage-take4 origin/libstorage-take4
curl -sSL https://dl.bintray.com/emccode/rexray/install | sh -
...add config file...
sudo rexray start
sudo su -
source /home/ubuntu/.profile
export ALLOW_PRIVILEGED=true
export LOG_LEVEL=5
export CLOUD_PROVIDER=aws
export KUBERNETES_PROVIDER=local
cd /home/ubuntu/work/src/k8s.io/kubernetes
hack/local-up-cluster.sh  lookup_pull.py    

root@ip-172-31-39-40:/home/ubuntu/work/src/k8s.io/kubernetes# hack/local-up-cluster.sh  lookup_pull.py
make: Entering directory '/home/ubuntu/work/src/k8s.io/kubernetes'
make[1]: Entering directory '/home/ubuntu/work/src/k8s.io/kubernetes'
+++ [1105 13:08:09] Generating bindata:
    /home/ubuntu/work/src/k8s.io/kubernetes/test/e2e/framework/gobindata_util.go
Cannot find go-bindata. Install with
  go get -u github.com/jteeuwen/go-bindata/go-bindata
  and make sure GOBIN is in the system PATH
../../../../../../test/e2e/framework/gobindata_util.go:19: running "../../../hack/update-bindata.sh": exit status 5
Makefile.generated_files:286: recipe for target '_output/bin/deepcopy-gen' failed
make[1]: *** [_output/bin/deepcopy-gen] Error 1
make[1]: Leaving directory '/home/ubuntu/work/src/k8s.io/kubernetes'
Makefile:287: recipe for target 'generated_files' failed
make: *** [generated_files] Error 2
make: Leaving directory '/home/ubuntu/work/src/k8s.io/kubernetes'
!!! Error in hack/local-up-cluster.sh:120
  'make -C "${KUBE_ROOT}" WHAT="cmd/kubectl cmd/hyperkube"' exited with status 2
Call stack:
  1: hack/local-up-cluster.sh:120 main(...)
Exiting with status 1
```

**Comment by kacole2** [comment]
@cantbewong i'm getting an error with REX-Ray after installation. really odd... got the build of k8 to start but i'm on a plane and it keeps disconnecting. will try again tomorrow

```
ubuntu@ip-172-31-33-51:~$ sudo rexray service start
● rexray.service - rexray
   Loaded: loaded (/etc/systemd/system/rexray.service; enabled; vendor preset: enabled)
   Active: active (running) since Mon 2016-11-07 04:00:28 UTC; 38s ago
 Main PID: 758 (rexray)
    Tasks: 7
   Memory: 57.5M
      CPU: 291ms
   CGroup: /system.slice/rexray.service
           └─758 /usr/bin/rexray start -f

Nov 07 04:00:29 ip-172-31-33-51 rexray[758]: ################################################################################
Nov 07 04:00:29 ip-172-31-33-51 rexray[758]: ##                                                                            ##
Nov 07 04:00:29 ip-172-31-33-51 rexray[758]: ##                  libStorage started  - 2016/11/07 04:00:29.773             ##
Nov 07 04:00:29 ip-172-31-33-51 rexray[758]: ##                                                                            ##
Nov 07 04:00:29 ip-172-31-33-51 rexray[758]: ##     endpoints:   tcp://127.0.0.1:7979                                      ##
Nov 07 04:00:29 ip-172-31-33-51 rexray[758]: ##                                                                            ##
Nov 07 04:00:29 ip-172-31-33-51 rexray[758]: ##      services:   name=ebs, driver=ebs                                      ##
Nov 07 04:00:29 ip-172-31-33-51 rexray[758]: ##                                                                            ##
Nov 07 04:00:29 ip-172-31-33-51 rexray[758]: ################################################################################
Nov 07 04:01:06 ip-172-31-33-51 systemd[1]: Started rexray.
ubuntu@ip-172-31-33-51:~$ sudo rexray volume ls
Oops, an error occured!

  listen tcp 127.0.0.1:7979: bind: address already in use

To correct the error please review:

  - Debug output by using the flag -l debug
  - The REX-ray website at https://github.com/emccode/rexray
  - The on↓ine he↓p be↓ow

List volumes
```

**Comment by kacole2** [comment]
config.yml is incorrect. please use this...

```
rexray:
  logLevel: debug
libstorage:
  host:     tcp://127.0.0.1:7979
  embedded: true
  service:  ebs
  server:
    endpoints:
      public:
        address: tcp://:7979
    services:
      ebs:
        driver: ebs
ebs:
  accessKey: #{$aws_akey}
  secretKey: #{$aws_skey}
```

**Comment by kacole2** [comment]
getting a 500 error when deploying a k8 pod.

```
ubuntu@ip-172-31-43-139:~/work/src/k8s.io/kubernetes$ cluster/kubectl.sh describe pod
Name:        mysql
Namespace:    default
Node:        127.0.0.1/127.0.0.1
Start Time:    Mon, 07 Nov 2016 18:01:11 +0000
Labels:        name=mysql
Status:        Pending
IP:
Controllers:    <none>
Containers:
  mysql:
    Container ID:
    Image:        mysql:5.6
    Image ID:
    Port:        3306/TCP
    State:        Waiting
      Reason:        ContainerCreating
    Ready:        False
    Restart Count:    0
    Volume Mounts:
      /var/lib/mysql from mysql-persistent-volume (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-0bmm6 (ro)
    Environment Variables:
      MYSQL_ROOT_PASSWORD:    yourpassword
Conditions:
  Type        Status
  Initialized     True
  Ready     False
  PodScheduled     True
Volumes:
  mysql-persistent-volume:
    Type:    LibStorage (a volume managed by LibStorage)
    Host:    http://127.0.0.1:7979
    Service:    ebs
    VolumeName:    mysql-1
    FSType:
    ReadOnly:    false
  default-token-0bmm6:
    Type:    Secret (a volume populated by a Secret)
    SecretName:    default-token-0bmm6
QoS Class:    BestEffort
Tolerations:    <none>
Events:
  FirstSeen    LastSeen    Count    From            SubObjectPath    Type        Reason        Message
  ---------    --------    -----    ----            -------------    --------    ------        -------
  55s        55s        1    {default-scheduler }            Normal        Scheduled    Successfully assigned mysql to 127.0.0.1
  55s        23s        7    {controller-manager }            Warning        FailedMount    Failed to attach volume "mysql-persistent-volume" on node "127.0.0.1" with: unexpected http status code 500
```

**Comment by cantbewong** [comment]
Will resubmit as a new PR

**Pull request closed without merging**